### PR TITLE
Ensure for a configurable origin in XCM (#6442)

### PR DESCRIPTION
**Pull request by serkul**
Rebase against XCM v3 changes.
Add new associated type, AdminOrigin, bounded by EnsureOrigin trait in XCM-pallet config. Replace ensure_root() with ensure_origin(). Use EnsureRoot<AccountId> in all implementations of XCM pallet to preserve the current behavior until Gov2 specific origins are implemented.

### CODE REVIEW

File: xcm/pallet-xcm/src/lib.rs
Comment (KiChjang) [inline xcm/pallet-xcm/src/lib.rs:235]: This is not actually what the `AdminOrigin` is supposed to do. It's an origin that is allowed to call privileged operations on the XCM pallet.

Comment (serkul) [reply xcm/pallet-xcm/src/lib.rs]: fixed the comment line.

Comment (serkul) [reply xcm/pallet-xcm/src/lib.rs]: For my own understanding, is it correct that, in future, `AdminOrigin` can include more origins and not only root?

Comment (KiChjang) [reply xcm/pallet-xcm/src/lib.rs]: Correct, that's the entire purpose of the issue.

**Review by KiChjang — APPROVED** [review]

**Comment by KiChjang** [comment]
Yes; check the CI failures and make sure they're resolved. Other than that, this is waiting for a review from members of @paritytech/polkadot-review.

**Comment by vstam1** [comment]
bot help

**Comment by vstam1** [comment]
bot fmt

**Pull request closed without merging**
